test/boost/filtering_test.cc

```diff
@@ -1176,3 +1176,155 @@ SEASTAR_TEST_CASE(test_filtering) {
+        
+        // Now we have all types of NULLs
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE flt > -999.9 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE dbl > -999.9 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE dec > -999.9 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE b > false ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE ti > -99 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE si > -999 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE i > -999 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE bi > -999 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE vi > -999 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE ip > '0.0.0.0' ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE u > 123e4567-e89b-12d3-a456-426655440000 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE tu > 123e4567-e89b-12d3-a456-426655440000 ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE t > '00:00:00' ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE d > '1970-01-01' ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE a > '0' ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+        {
+            auto msg = e.execute_cql("SELECT * FROM cf WHERE txt > '0' ALLOW FILTERING;").get0();
+            assert_that(msg).is_rows().is_empty();
+        }
+    });
+}
+
+SEASTAR_TEST_CASE(test_filtering_lt_not_capturing_nulls) {
```

Comment:
I think it would be better to just have *one* test, which will run a loop on several operators. There are two benefits to my idea:
1. Less code duplication, and less chance of accidental difference between the duplicated versions.
2. Easy to check additional operators - what about testing <=, >=, =, != ?

Reply:
Yes, there's way too much duplication here.  Also, I'm not sure we need to cover every single operator/type combination -- that's just bloating the runtime without much additional confidence.

Reply:
I reduced duplication, but kept only the "<" and ">" cases. Would you like to see any other operator? (honestly, I don't know)

Reply:
I think `>` and `<` are enough to catch any bugs that creep in.  (I also think that trying every single type is superfluous and stale-bait, but that's not enough to hold back this PR.)

Reply:
It depends how much time each case takes. If each case takes a millisecond, I don't see any reason not to do it. But I don't insist. I'll accept this as-is.